pynamodb/tests/test_signals.py

```diff
@@ -0,0 +1,50 @@
+import json
+
+import mock
+import pytest
+import requests
+
+from pynamodb.connection import Connection
+from pynamodb.connection.signals import pre_boto_send, post_boto_send
+
+try:
+    import blinker
+except ImportError:
+    blinker = None
+
+pytestmark = pytest.mark.skipif(
+    blinker is None,
+    reason='Signals require the blinker library.'
+)
+
+@mock.patch('pynamodb.connection.Connection.session')
+@mock.patch('pynamodb.connection.Connection.requests_session')
+@mock.patch('pynamodb.connection.base.uuid')
+def test_signal(mock_uuid, requests_session_mock, session_mock):
+    pre_recorded = []
+    post_recorded = []
+    UUID = '123-abc'
+
+    def record_pre_boto_send(sender, operation_name, table_name, req_uuid):
```

Comment:
I have it set up to pass in the Connection class as the emitter, which was the simplest way to implement this.

Unfortunately, the Connection class does give away a lot of information. Also, it is not very intuitive that you have to have this as an argument.

I have another implementation where we have a wrapper around the `pre_boto_send.connect` call so you don't have to handle the sender, but it requires more bookkeeping and the code got complicated. thought I push the simple change for now and see what other people thought.